benchmark/benchmark_generate_cpu.py

```diff
@@ -0,0 +1,110 @@
+import argparse
+import os
+
+os.environ["OMP_NUM_THREADS"] = "1"
+os.environ["MKL_NUM_THREADS"] = "1"
+import time
+import warnings
+
+warnings.filterwarnings("ignore")
+
+import torch
+
+torch.set_num_threads(8)
+from torch import nn
+
+from transformers import AutoConfig, AutoModelForCausalLM
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(add_help=True)
+    parser.add_argument(
+        "--model",
+        type=str,
+        required=True,
+    )
+    parser.add_argument(
+        "--num_codebooks",
+        type=int,
+        default=None,
+    )
+    parser.add_argument(
+        "--in_group_size",
+        type=int,
+        default=None,
+    )
+    parser.add_argument(
+        "--nbits_per_codebook",
+        type=int,
+        default=None,
+    )
+    parser.add_argument(
+        "--warmup_iters",
+        type=int,
+        default=1,
+        help="Number of warmup iterations.",
+    )
+    parser.add_argument(
+        "--benchmark_iters",
+        type=int,
+        default=3,
+        help="Number of benchmark iterations.",
+    )
+    parser.add_argument(
+        "--input_length",
+        type=int,
+        default=1,
+        help="Input length.",
+    )
+    parser.add_argument(
+        "--output_length",
+        type=int,
+        default=128,
+        help="Output length.",
+    )
+    args = parser.parse_args()
+
+    device = "cpu"
+
+    config = AutoConfig.from_pretrained(args.model, trust_remote_code=True, torch_dtype=torch.float32)
+    if args.num_codebooks is not None:
+        config.aqlm["num_codebooks"] = args.num_codebooks
+    if args.in_group_size is not None:
+        config.aqlm["in_group_size"] = args.in_group_size
+    if args.nbits_per_codebook is not None:
+        config.aqlm["nbits_per_codebook"] = args.nbits_per_codebook
+
+    real_num_layers = config.num_hidden_layers
+    if "meta-llama" in args.model:
+        config.num_hidden_layers = 1
+    aqlm_model = AutoModelForCausalLM.from_config(config, trust_remote_code=True, torch_dtype=torch.float32)
+
+    if "meta-llama" in args.model:
```

Comment:
Can't find an else statement. consider: else assert NotImplementedError(...)?

Reply:
It's more complicated. It's meant to differentiate between quantised and unquantized models like that.